### PR TITLE
[WIP] clean up onboard, require onboarding when no user registered, add svgs

### LINKED ISSUE #98: Onboarding screens
Need to implement maybe 3-4 on boarding screens explaining the app. This will greatly improve UX

Technical implementation is already complete just need relevant copy/illustrations

To do:

- [ ] Copy for 3-4 screens
- [ ] Find 3-4 illustrations [something like this](https://undraw.co/illustrations)

### PR DESCRIPTION
**Pull request by oscarmanderj**
**Problem:**
No explanation about the app

**Solution:**
onboarding screens on first boot-up explaining app.

COPY IS IN DRAFT, another PR will complete this

See on this file on this branch for the copy:
https://github.com/voteflux/voting_app/blob/feature/onboarding/lib/ui/views/onboarding_view.dart

closes #98

### CODE REVIEW

**Comment by oscarmanderj** [comment]
Not sure about images... Some are good, some not so much.

**Comment by oscarmanderj** [comment]
First time opening app -> 
onboarding flow ( currently 5 slides with brief overview off app) ->
sign up page (just name for now, until we have final implementation) ->
All bills page

Next time opening app -> 
All bills page

**Comment by XertroV** [comment]
So, is this right?

first time opening app -> slides -> close app
open app -> all bills

is there a "go straight to issues" button on the signup page?

**Comment by oscarmanderj** [comment]
after slides they enter their name (which is locally stored, but can remove) ->  all bills page (no need to close app)

Open app ->  all bills (it's the 'home page')

No 'straight to issues' button - All users have to go through signup, which generates a private key, which they can use to vote.

I think we should push all users through the 'private key' creation, even if it's just in the background. The current code assumes this, otherwise will have to write more UI code, just to cover the 'I want to signup now'.

**Comment by XertroV** [comment]
> No 'straight to issues' button - All users have to go through signup, which generates a private key, which they can use to vote.

Re privkey, is it not already in the background? Like it's currently connected to nav/signup logic?

> I think we should push all users through the 'private key' creation, even if it's just in the background. The current code assumes this, otherwise will have to write more UI code, just to cover the 'I want to signup now'.

Assumes the user goes through signup, you mean?

This feels like a red flag re design. If we have crypto logic mixed in with navigation logic, that doesn't feel great. Are the reasons this has happened not going to cause problems for us in the future?

particularly: a users state can be in _any_  configuration; or at least we should be capable of dealing with it. We can have a privkey with no name (like so a user knows we're not gathering telemetry), registration is something that will expire and need to be renewed, etc.

I'd be sorta surprised if it required many changes; the voting wallets I've done have been pretty thin implementation wrappers, even with actions and stuff.

File: lib/ui/views/onboarding_view.dart [hunk 1]
```diff
@@ -24,58 +26,75 @@ class _OnBoardingViewState extends State<OnBoardingView> {
...
+                      _buildWalkThrough(context, 'undraw_welcome_3gvl.svg',
+                          "Welcome to DigiPol! Where digital direct democracy starts."),
+                      _buildWalkThrough(context, 'undraw_process_e90d.svg',
+                          "DigiPol collates all the current bills in the Australian goverment for you."),
+                      _buildWalkThrough(context, 'undraw_voting_nvu7.svg',
+                          "So you can stay informed and vote directly on what matters most."),
+                      _buildWalkThrough(context, 'undraw_new_ideas_jdea.svg',
+                          "You can even vote on current issues created by the community or submit your own."),
+                      _buildWalkThrough(context, 'undraw_ethereum_fb7n.svg',
+                          "We don't store any of your information, and all votes are verified by the ethereum blockchain."),
```
Comment (XertroV) [inline lib/ui/views/onboarding_view.dart:45]: Um. Who's "we"

File: lib/ui/views/onboarding_view.dart [hunk 2]
```diff
@@ -24,58 +26,75 @@ class _OnBoardingViewState extends State<OnBoardingView> {
...
+                  child: PageView(
+                    children: <Widget>[
+                      _buildWalkThrough(context, 'undraw_welcome_3gvl.svg',
+                          "Welcome to DigiPol! Where digital direct democracy starts."),
+                      _buildWalkThrough(context, 'undraw_process_e90d.svg',
+                          "DigiPol collates all the current bills in the Australian goverment for you."),
+                      _buildWalkThrough(context, 'undraw_voting_nvu7.svg',
+                          "So you can stay informed and vote directly on what matters most."),
+                      _buildWalkThrough(context, 'undraw_new_ideas_jdea.svg',
+                          "You can even vote on current issues created by the community or submit your own."),
```
Comment (XertroV) [inline lib/ui/views/onboarding_view.dart:43]: if we do a release it has to be accurate

**Comment by XertroV** [comment]
Re the design issues above (which we do need to fix, the earlier the better) - they weren't added in this PR, so they're not a reason not to merge it. However, the copy on the slides needs work. 

Master = next release in our model => (implying) releasable, so unless stuff like user-submittable issues is in there, it can't go in like that. Even just `(in development)` or something is enough to add (provided that were true). In this case, just commenting that one is fine.

Making a claim about user data is serious, and we should take it very seriously. We're doing **voting**.

I suggest we change the info one to mention no user identifiable info is on the blockchain, another slide makes a comment about user info (okay to comment this or leave a todo), and we take out the user generated one entirely; though up to you if we replace it or not.

Also, I don't think the copy about "what matters most" fits here - digipol is DD not IBDD; you vote with 1 vote on everything or some subset of your choosing. Not really "what matters most". something like "make your opinion known *when* it matters most" makes far more sense.

File: lib/ui/views/onboarding_view.dart (hunk 1 above)
Comment (oscarmanderj) [reply lib/ui/views/onboarding_view.dart:45]: All copy was just place holder really. This could  just be changed to 'digipol'

Comment (XertroV) [reply lib/ui/views/onboarding_view.dart:45]: I shouldn't have said that, and instead said something clearer: you're making a strong promise on behalf of more than just yourself, and ambiguously at that. Isn't a vote 'their information' as such?

Comment (oscarmanderj) [reply lib/ui/views/onboarding_view.dart:45]: I was trying to just explain simply what happens with any data (voting, names etc). I agree, too ambiguous. What is the best wording for this?

**Comment by oscarmanderj** [comment]
re: design issues 

Private key gets generated after they put their name in and hit the signup button. The logic is not coupled together, it is just called at the same time in that method. Private key can be generated at some other point.

If user can traverse app without private key, only other change would be to conditionally remove vote widget. Which could be replaced with a 'sign up to vote'.

**Comment by oscarmanderj** [comment]
Made copy changes - I will make change user state changes in another pr

**Comment by XertroV** [comment]
> If user can traverse app without private key, only other change would be to conditionally remove vote widget. Which could be replaced with a 'sign up to vote'.

Yeah, that's great.

> Private key gets generated after they put their name in and hit the signup button. The logic is not coupled together, it is just called at the same time in that method. Private key can be generated at some other point.

cool. it's not too important when we generate it atm; if we do it rigorously like a proper crypto wallet we need a dedicated flow anyway b/c backups, etc. cross that bridge later

re copy: I would be much much lazier:

"(draft) something something blockchain security, identity not public, etc"

if you don't want to do copy in a PR, make it very obvious it's not copy. Since we're in "beta" (not that that means anything), this is okay, because:

1. if it's public it's obvious it's a draft and not reflecting reality
2. it's hard to miss

That's okay because there are no consequences besides needing to do the thing we would need to do anything.

File: lib/ui/views/onboarding_view.dart [hunk 3]
```diff
@@ -14,68 +16,82 @@ class _OnBoardingViewState extends State<OnBoardingView> {
...
+                  child: PageView(
+                    children: <Widget>[
+                      _buildWalkThrough(context, 'undraw_Hello_qnas.svg',
+                          "Welcome to DigiPol! Where digital direct democracy starts."),
+                      _buildWalkThrough(context, 'undraw_process_e90d.svg',
+                          "DigiPol collates all the current bills in the Australian goverment for you."),
+                      _buildWalkThrough(context, 'undraw_voting_nvu7.svg',
+                          "So you can stay informed & make your opinion known when it matters most."),
+                      _buildWalkThrough(context, 'undraw_new_ideas_jdea.svg',
+                          "You can even vote on current issues created by the community or submit your own (in development)."),
```
Comment (XertroV) [inline lib/ui/views/onboarding_view.dart:40]: It's not in development. We don't even know HOW we'll do it. We need terms of service, research into user-content in app, moderation standards, etc, etc, etc, etc, etc

Unless you are going to take those responsibilities, don't put the text down. It doesn't matter here, just say:

"we're hoping to add this in the future" or something!

File: lib/ui/views/onboarding_view.dart [hunk 4]
```diff
@@ -14,68 +16,82 @@ class _OnBoardingViewState extends State<OnBoardingView> {
...
+                      _buildWalkThrough(context, 'undraw_ethereum_fb7n.svg',
+                          "None of your personal information is stored, and all votes are verified by the ethereum blockchain."),
```
Comment (XertroV) [inline lib/ui/views/onboarding_view.dart:42]: fix

File: lib/ui/views/onboarding_view.dart [hunk 5]
```diff
@@ -14,68 +16,82 @@ class _OnBoardingViewState extends State<OnBoardingView> {
...
+              Flexible(
+                child: Container(
+                  child: PageView(
+                    children: <Widget>[
+                      _buildWalkThrough(context, 'undraw_Hello_qnas.svg',
+                          "Welcome to DigiPol! Where digital direct democracy starts."),
+                      _buildWalkThrough(context, 'undraw_process_e90d.svg',
+                          "DigiPol collates all the current bills in the Australian goverment for you."),
+                      _buildWalkThrough(context, 'undraw_voting_nvu7.svg',
+                          "So you can stay informed & make your opinion known when it matters most."),
```
Comment (XertroV) [inline lib/ui/views/onboarding_view.dart:38]: fix

File: lib/ui/views/onboarding_view.dart [hunk 6]
```diff
@@ -14,68 +16,82 @@ class _OnBoardingViewState extends State<OnBoardingView> {
   void initState() {
     super.initState();
     currentIndexPage = 0;
-    pageLength = 3;
+    pageLength = 6;
```
Comment (XertroV) [inline lib/ui/views/onboarding_view.dart:19]: don't hardcode, declare array of pages elsewhere. use `len()` or `.length` or whatever. you should be able to comment an entry from the array and not break the app.

File: lib/ui/views/onboarding_view.dart (hunk 1 above)
Comment (XertroV) [reply lib/ui/views/onboarding_view.dart]: If you remove copy as part of this PR, it doesn't matter.

File: lib/ui/views/onboarding_view.dart [hunk 7]
```diff
@@ -9,73 +11,104 @@ class OnBoardingView extends StatefulWidget {
...
+          'None of your personal information is stored, and all votes are verified by the ethereum blockchain.'
+    }
+  ];
 
   @override
   void initState() {
     super.initState();
     currentIndexPage = 0;
-    pageLength = 3;
+    pageLength = pages.length + 1;
```
Comment (XertroV) [inline lib/ui/views/onboarding_view.dart:44]: presumably this is okay

**Comment by XertroV** [comment]
@okdaithi - I think your comments are good, but some aren't related to this PR. We should try to separate things that are issues with the PR itself since they're the only things that should prevent merging.

@oscarmanderj - I think you've probably put too much in the *scope* of this PR. This PR should probably just be about **one thing:** adding *slides*.

- **if you keep copy in there**, the copy will block merging, **you have currently kept copy in there**. This is why I mentioned the "(draft) something something blockchain something" thing. That's fine, because it's OBVIOUS copy isn't a part of this. If a user sees that they will have like zero expectations because of it. Right now, not only is not obvious (saying something in your PR comment doesn't count - we don't merge that), it contradicts the PR comment b/c of this.
- if you refer to onboarding besides just integrating with existing flow, onboarding stuff will block merging.

If you want to have a separate PR for copy and leave this copy how it is - fine, just set this PR as the target of the other PR and when the other PR is done we can come back to this one.

Even when you say: 

> onboarding screens on first boot-up explaining app.

Are they onboarding screens or first-run screens? If those things are _currently_ the same, that doesn't necessarily pose an issue, but if you say they're both then there's more discussion. They seem like first-run screens, not onboarding screens.

**Comment by okdaithi** [comment]
@oscarmanderj I've put some copy and layout ideas in this file.

https://docs.google.com/presentation/d/1gxOrGpmOjTf10Ies37-hREUpDlAC_YhiNnBLYkugX9s/edit?usp=sharing

I'll start looking for some additional graphics too. Where did you get the current images?

**Comment by XertroV** [comment]
Just on the note of images, while I remember, we need to note down sources of images and make sure we e.g. comply with creative commons requirements or w/e. don't want to get the app pulled b/c one image was copied illegitimately.

**Comment by oscarmanderj** [comment]
Yes am well aware of this. All of these came from https://undraw.co/ which is free to use even commercially

**Comment by okdaithi** [comment]
> 
> 
> Yes am well aware of this. All of these came from https://undraw.co/ which is free to use even commercially

I went through undraw catalogue and found some more images which I've inserted into the mocked up slides above.

All pngs and svgs in this folder:
https://drive.google.com/drive/folders/1ejw1y_Z6SPINpYJmAOItD_xv8nNK33kX?usp=sharing

**Comment by oscarmanderj** [comment]
If we can merge this, I'll do another PR for on-boarding copy and images.

**Comment by XertroV** [comment]
discussed in dev chat

**Pull request merged**